tests/conftest.py

```diff
@@ -219,44 +183,76 @@ def user_teardown():
             session.execute(table.delete())
 
 
+def encoded_jwt(private_key, user):
+    """
+    Return an example JWT containing the claims and encoded with the private
+    key.
+
+    Args:
+        private_key (str): private key
+        user (userdatamodel.models.User): user object
+
+    Return:
+        str: JWT containing claims encoded with private key
+    """
+    kid = JWT_KEYPAIR_FILES.keys()[0]
+    scopes = ['openid']
+    return generate_signed_access_token(
+        kid, private_key, user, 3600, scopes, forced_exp_time=None)
+
+
+def create_user_header(app, request, pg_driver, username):
+    private_key = utils.read_file('resources/keys/test_private_key.pem')
+
+    user_driver = SQLAlchemyDriver(PSQL_USER_DB_CONNECTION)
+    with user_driver.session as s:
+        user = s.query(usermd.User).filter_by(username=username).first()
+        token = encoded_jwt(private_key, user)
+        return {'Authorization': 'bearer ' + token}
+
+
 @pytest.fixture()
-def submitter(app, request):
-    def build_header(path, method, role='submitter'):
-        auth = get_auth(role + 'accesskey', role, 'submission')
-        environ = make_test_environ_builder(app, path=path, method=method)
-        request = environ.get_request()
-        request.headers = dict(request.headers)
-        auth.__call__(request)
-        return request.headers
+def submitter(app, request, pg_driver):
+    return create_user_header(app, request, pg_driver, 'submitter')
 
-    return build_header
 
+@pytest.fixture()
+def admin(app, request, pg_driver):
```

Comment:
I see the request parameter getting passed into these fixtures without ever being used. Since they are functionally scoped and reading a file did you mean to add a finalizer?

Reply:
thanks for catching that, yeah don't need app and request. 
pg_driver is there to utilize it's database setup and cleanup because those roles fixture depends on those users exist in the db